node/src/daemon_json/json_ledger.rs

```diff
@@ -61,6 +65,36 @@ impl Ledger {
             }
         }
     }
+
+    /// Typically a ledger is identified by its hash, but if hash is
+    /// not given in config, we construct a name by hashing the ledger
+    /// config so that we can recognise a ledger we already have
+    /// cached without reconstructing it (which is expensive).
+    /// Note that the name contains some constants - the purpose of this
+    /// is to invalidate all caches when these constants change.
+    pub fn ledger_name(&self) -> String {
```

Comment:
@Sventimir does OCaml perform this same hashing? if that is the case please include a permalink to the original implementation.

Reply:
I updated the last commit with the the following link:
https://github.com/MinaProtocol/mina/blob/develop/src/lib/genesis_ledger_helper/genesis_ledger_helper.ml#L105